test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,397 @@
+// +build token_propagation_elasticsearch
+
+package e2e
+
+import (
+	"bytes"
+	goctx "context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/uber/jaeger-client-go/config"
+	"golang.org/x/net/html"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis"
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	esv1 "github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1"
+)
+
+// Test parameters
+const name = "token-prop"
+const username = "user-test-token"
+const password = "any"
+const collectorPodImageName = "jaeger-collector"
+const testServiceName = "token-propagation"
+
+type TokenPropagationTestSuite struct {
+	suite.Suite
+	exampleJaeger        *v1.Jaeger
+	queryName            string
+	collectorName        string
+	queryServiceEndPoint string
+	host                 string
+}
+
+func (suite *TokenPropagationTestSuite) SetupSuite() {
+	t = suite.T()
+	if !isOpenShift(t) {
+		t.Skipf("Test %s is currently supported only on OpenShift because es-operator runs only on OpenShift\n", t.Name())
+	}
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &v1.JaegerList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+	}))
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &esv1.ElasticsearchList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Elasticsearch",
+			APIVersion: "logging.openshift.io/v1",
+		},
+	}))
+	var err error
+	ctx, err = prepare(t)
+	if err != nil {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+	addToFrameworkSchemeForSmokeTests(t)
+
+	suite.deployJaegerWithPropagationEnabled()
+
+}
+
+func (suite *TokenPropagationTestSuite) TearDownSuite() {
+	// undeployJaegerInstance(suite.exampleJaeger)
+	// handleSuiteTearDown()
+}
+
+func (suite *TokenPropagationTestSuite) TestTokenPropagationNoToken() {
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		req, err := http.NewRequest(http.MethodGet, suite.queryServiceEndPoint, nil)
```

Comment:
Check that error is nil, and return early in case of errors (`return false, err`). Otherwise, it will fail in the next line.

Reply:
Looks like this wasn't done yet.